sirepo/template/hdf5_util.py

```diff
@@ -88,3 +90,14 @@ def lineplot(self, plot_attrs):
                 x_label=x.label,
             ),
         )
+
+    @contextlib.contextmanager
+    def read_while_writing(self, retries=10, timeout=3):
+        for _ in range(retries):
+            try:
+                with h5py.File(self.filename, "r") as p:
+                    yield p
+                    return
+            except (BlockingIOError, IOError, KeyError) as e:
+                time.sleep(timeout)
+        raise AssertionError(f"{self.filename} is unavailable")
```

Comment:
I think this only handles the not-found case. When I run the test read program by itself, it doesn't retry:
```
python read.py x.h5 check_key
...
KeyError: "Unable to open object (object 'this_key_wont_exist_at_first' doesn't exist)"
...
RuntimeError: generator didn't stop after throw()
```

I think this is related to this:
https://stackoverflow.com/questions/34775099/why-does-contextmanager-throws-a-runtime-error-generator-didnt-stop-after-thro

Instead, you should pass a callback to read_while_writing() which gets executed within the try/except.